strawberryfields/api_client.py

```diff
@@ -0,0 +1,546 @@
+
+        Args:
+            job_id (int): The ID of an object to be retrieved.
+        """
+        if "GET" not in self.resource.SUPPORTED_METHODS:
+            raise MethodNotSupportedException("GET method on this resource is not supported")
+
+        response = self.client.get(self.join_path(str(job_id)))
+        self.handle_response(response)
+
+    def create(self, **params):
+        """
+        Attempts to create a new instance of a resource by sending a POST
+        request to the appropriate endpoint.
+
+        Args:
+            **params: Arbitrary parameters to be passed on to the POST request.
+        """
+        if "POST" not in self.resource.SUPPORTED_METHODS:
+            raise MethodNotSupportedException("POST method on this resource is not supported")
+
+        if self.resource.id:
+            raise ObjectAlreadyCreatedException("ID must be None when calling create")
+
+        response = self.client.post(self.resource.PATH, params)
+
+        self.handle_response(response)
+
+    def handle_response(self, response):
+        """
+        Store the status code on the manager object and handle the response
+        based on the status code.
+
+        Args:
+            response (requests.Response): A response object to be parsed.
+        """
+        self.http_status_code = response.status_code
+        if response.status_code in (200, 201):
+            self.handle_success_response(response)
+        else:
+            self.handle_error_response(response)
+
+    def handle_success_response(self, response):
+        """
+        Handles a successful response by refreshing the instance fields.
+
+        Args:
+            response (requests.Response): A response object to be parsed.
+        """
+        self.refresh_data(response.json())
+
+    def handle_error_response(self, response):
+        """
+        Handles an error response that is returned by the server.
+
+        Args:
+            response (requests.Response): A response object to be parsed.
+        """
+
+        # TODO: Improve error messaging and parse the actual error output (json).
+
+        if response.status_code in (400, 409):
+            warnings.warn(
+                "The server did not accept the request, and returned an error "
+                "({}: {}).".format(response.status_code, response.text),
+                UserWarning,
+            )
+        elif response.status_code == 401:
+            warnings.warn(
+                "The server did not accept the request due to an authentication error "
+                "({}: {}).".format(response.status_code, response.text),
+                UserWarning,
+            )
+        elif response.status_code in (500, 503, 504):
+            warnings.warn(
+                "The client encountered an unexpected temporary server error "
+                "({}: {}).".format(response.status_code, response.text),
+                UserWarning,
+            )
```

Comment:
what about other errors, e.g., 404?

Reply:
I am adding 404 error handling, as well as a catchall. This section of the code will be refactored in the future as well for better error handling, it's not really doing anything right now.